source/elements/oneDAL/source/glossary.rst

```diff
@@ -0,0 +1,117 @@
+.. _glossary:
+
+=========
+Glossary
+=========
+
+Machine learning terms
+======================
+
+.. glossary::
+    :sorted:
+
+    Attribute
+        A particular property or quality of a real object or an event.
+        Has a defined type and domain.
+
+    Classification
+        A supervised machine-learning problem to assign categorical :term:`labels<Label>`
+        to :term:`feature vectors<Feature vector>` describing input objects.
+
+    Dataset
+        A collection of :term:`instances<Instance>`.
+
+    Feature
+        A pair of :term:`attribute` information and a corresponding value.
+
+    Feature vector
+        A vector that encodes information about single real object or an event.
+        Contains at least one :term:`feature`.
+
+    Inference
+        A process of obtaining outputs from the :term:`model` created on the :term:`training` stage
+
+    Instance
+        A vector of at least one :term:`feature` value and zero or more :term:`label<Label>` values.
+        Each of :term:`labels<Label>` and :term:`features<Feature>` can encode different objects or events.
+
+    Label
+        :term:`Feature` of some object or event which is present on the :term:`training` stage but
+        unavaliable on the :term:`inference` stage.
+
+    Model
```

Comment:
general question: do we rely on some literature that introduces the terms or define our self

Reply:
I formulate it by myself, but with accordance to external resources. I use [this](https://ml-cheatsheet.readthedocs.io/en/latest/glossary.html) and [this](https://developers.google.com/machine-learning/glossary) links and Peter Flach's book to check the terms.

Reply:
option to think: if there is a well recognized a book/source with thedefs we might rely on it and then refer that we use this source as the base for our ml glossary in the volumes necessary for the lib

Reply:
Yes, it might be good and probably will give a "weight" to our terminology in the reader's eyes. 
But I really don't know, can we call the resources which I list somewhere below as a "well recognized"? How to measure it? I mostly rely on the **number** of resources and their **difference** to formulate more or less "objective" definitions.
Another thing is that most of the books have very long, "continuous" terms, which need to be reformulated shortly. Can we refer to a book if we mostly reformulated the term which is described in it?
 @andreynv @rlnx can you please name some well-recognized sources which I should also consider?

Reply:
Mentioned one somewhere in the other comment and let you know, if I have another